### PR TITLE
Update to Substrate master

### PR DESCRIPTION
**Pull request by gavofyork**
This introduces a new type `CollatorId`, currently just `SessionKey` but which would forseeably change to its own thing. It seems to work like this (despite there being a lot of the new-incompatible `AccountId` replaced). No idea if it does anything sensible, though.

### CODE REVIEW

File: collator/src/lib.rs [hunk 1]
```diff
@@ -127,11 +127,6 @@ pub trait RelayChainContext {
 	fn unrouted_egress(&self, id: ParaId) -> Self::FutureEgress;
 }
 
-fn key_to_account_id(key: &ed25519::Pair) -> AccountId {
```
Comment (gavofyork) [inline collator/src/lib.rs]: Note that this (and functions like it) are broken by design.

**Comment by gavofyork** [comment]
Added specific crypto types `parachain::CollatorId` and `parachain::ValidatorId`. For now they are equivalent to `SessionKey`, but we should talk about what we want to do with them soon.

File: network/src/collator_pool.rs [hunk 2]
```diff
@@ -253,19 +252,19 @@ mod tests {
...
 		let (tx2, rx2) = oneshot::channel();
 		pool.await_collation(relay_parent, para_id, tx1);
 		pool.await_collation(relay_parent, para_id, tx2);
-		pool.on_collation(primary, relay_parent, Collation {
+		pool.on_collation(primary.clone(), relay_parent, Collation {
 			receipt: CandidateReceipt {
 				parachain_index: para_id,
 				collator: primary.into(),
-				signature: H512::from([2; 64]).into(),
+				signature: Default::default(),//H512::from([2; 64]).into(),
```
Comment (gavofyork) [inline network/src/collator_pool.rs]: Looks like this value isn't checked?

File: primitives/src/parachain.rs [hunk 3]
```diff
@@ -89,16 +102,17 @@ pub struct Extrinsic {
 
 /// Candidate receipt type.
 #[derive(PartialEq, Eq, Clone, Encode, Decode)]
-#[cfg_attr(feature = "std", derive(Serialize, Deserialize, Debug))]
-#[cfg_attr(feature = "std", serde(rename_all = "camelCase"))]
-#[cfg_attr(feature = "std", serde(deny_unknown_fields))]
+//#[cfg_attr(feature = "std", derive(Serialize, Deserialize, Debug))]
+#[cfg_attr(feature = "std", derive(Debug))]
+//#[cfg_attr(feature = "std", serde(rename_all = "camelCase"))]
```
Comment (rphmeier) [inline primitives/src/parachain.rs]: commented code

File: service/src/lib.rs [hunk 4]
```diff
@@ -46,10 +43,10 @@ pub mod chain_spec;
 
 use std::sync::Arc;
 use std::time::Duration;
-use polkadot_primitives::{parachain, AccountId, Block, Hash, BlockId};
+use polkadot_primitives::{parachain, parachain::CollatorId, Block, Hash, BlockId};
 use polkadot_runtime::{GenesisConfig, RuntimeApi};
 use polkadot_network::gossip::{self as network_gossip, Known};
-use primitives::ed25519;
+use primitives::{Pair, ed25519};
```
Comment (rphmeier) [inline service/src/lib.rs]: i'm surprised that `Pair` would be exported at the root if there is more than one kind of crypto. for readability we should explicitly import from some submodule. I have no idea what kind of crypto this keypair is for.

File: test-parachains/adder/collator/src/main.rs [hunk 5]
```diff
@@ -31,7 +31,7 @@ use std::collections::HashMap;
 use std::sync::Arc;
 
 use adder::{HeadData as AdderHead, BlockData as AdderBody};
-use substrate_primitives::ed25519::Pair;
+use substrate_primitives::{Pair as _Pair, ed25519::Pair};
```
Comment (rphmeier) [inline test-parachains/adder/collator/src/main.rs]: What is `_Pair` used for here?

File: service/src/lib.rs (hunk 4 above)
Comment (gavofyork) [reply service/src/lib.rs]: It's the trait.

File: test-parachains/adder/collator/src/main.rs (hunk 5 above)
Comment (gavofyork) [reply test-parachains/adder/collator/src/main.rs]: Trait

File: runtime/Cargo.toml [hunk 6]
```diff
@@ -14,7 +14,7 @@ polkadot-primitives = { path = "../primitives", default-features = false }
 parity-codec = { version = "3.0", default-features = false }
 parity-codec-derive = { version = "3.0", default-features = false }
 substrate-serializer = { git = "https://github.com/paritytech/substrate", default-features = false }
-sr-std = { git = "https://github.com/paritytech/substrate", default-features = false }
+sr-std = { git = "https://github.com/paritytech/substrate", commit = "45824913c980bb1ba3963f9bba67775a507d8624", default-features = false }
```
Comment (bkchr) [inline runtime/Cargo.toml]:
```suggestion
sr-std = { git = "https://github.com/paritytech/substrate", rev = "45824913c980bb1ba3963f9bba67775a507d8624", default-features = false }
```

`commit` is no valid keyword here (Cargo probably ignores it)

File: test-parachains/adder/collator/src/main.rs (hunk 5 above)
Comment (bkchr) [reply test-parachains/adder/collator/src/main.rs]: Why not `PairT` as we do it with `BlockT` or `HeaderT`.

**Pull request merged**